core/src/main/java/com/radixdlt/consensus/PendingVotes.java

```diff
@@ -124,17 +124,32 @@ private void checkForDivergentVertexExecution(Vote vote) {
       final var otherVertexId = otherVote.getValue().proposedHeader().getVertexId();
       final var otherLedgerHeader = otherVote.getValue().proposedHeader().getLedgerHeader();
       if (voteVertexId.equals(otherVertexId) && !voteLedgerHeader.equals(otherLedgerHeader)) {
-        log.warn(
-            "Divergent vertex execution detected! An incoming vote (from {}) for vertex {} claims"
-                + " the following resultant ledger header: {}, while validator {} thinks that the"
-                + " resultant ledger header is {}. [this_vote={}, other_vote={}]",
-            vote.getAuthor(),
-            voteVertexId,
-            voteLedgerHeader,
-            otherVote.getKey(),
-            otherLedgerHeader,
-            vote,
-            otherVote);
+        if (!voteLedgerHeader
+            .nextProtocolVersion()
+            .equals(otherLedgerHeader.nextProtocolVersion())) {
+          log.info(
+              """
+              Received BFT votes for conflicting protocol update headers.
+              Validator {} enacts {}, while validator {} enacts {}.
+              This indicates that one or more validators run an outdated node version.
```

Comment:
This isn't always true - when it comes to the next update, it could be that the other ledger header is the one with the protocol version and therefore the one that has been updated.

I suggest splitting the cases of:
* Ours has a next protocol version, their's has no protocol version => Likely they are running an outdated version, all good. Probably a debug, not even an info(!)
* Theirs has protocol version, ours doesn't match => Possibly we're running an outdated version? Or potentially it's just more generally a malicious node messing with us. We probably want a variant of the `warn` message, with some initial emphasis that we might be running an outdated version.

Other issues:
* Also I notice the `checkForDivergentVertexExecution(vote)` happens before we even check the vote is from someone in the validator set. We should probably move it a step later.
* These logs may need limiting in some manner? Otherwise we could have a lot of spam if we're logging for every vote in every round! Although I guess ideally this will only happen for one round before the other validators crash... So maybe this is okay.
* If the validator set actually commits the next protocol update (which I imagine will be the standard here - most of the validator set will have updated!), we're going to crash anyway when trying to sync the last transactions of the previous epoch, with a divergent resolved proof. Can we double check we do do this assertion when we sync, and that we have an explicit log for this case before we crash / panic? e.g. something along the lines of "FATAL: The validator set have enacted anemone, but this node does not know about anemone." _or_ "FATAL: The validator set have enacted anemone, but this node does not expect anemone to trigger at this point.". I'm sure we do have that check, but I just want to sanity check.

Reply:
> This isn't always true - when it comes to the next update, it could be that the other ledger header is the one with the protocol version and therefore the one that has been updated.

Well, this is _almost_ always true, IMO true-enough for this log message :smile:
If there are two votes that produce distinct `next_protcol_version` then they must use different `ProtocolConfig`s, of which one must be newer - unless there's a bug or someone has deliberately misconfigured their nodes (which could be considered "outdated"), but there's no need to provide such details here. The message just says that "one or more validators [in the validator set] run an outdated node", it doesn't specify if it's this node or some other.

There isn't really distinction between "ours" and "theirs". These are just two votes from two validators in the validator set. Although maybe it's worth adding it and extending the messages, as you suggest :thinking: 


> Also I notice the checkForDivergentVertexExecution(vote) happens before we even check the vote is from someone in the validator set. We should probably move it a step later.

Hmh seems that we actually check that twice (first time in `BFTEventStatelessVerifier`).
But agreed, I'll move it a step later, which is more correct from `PendingVotes`s perspective.

> These logs may need limiting in some manner? Otherwise we could have a lot of spam if we're logging for every vote in every round! Although I guess ideally this will only happen for one round before the other validators crash... So maybe this is okay.

Good point, I'll make sure this isn't too spammy.

> If the validator set actually commits ...  but I just want to sanity check.

We have:
`resultant protocol update at version {} differs from the proof ({:?} != {:?})`.
Which isn't too friendly, I'll update it.

Reply:
> There isn't really distinction between "ours" and "theirs". These are just two votes from two validators in the validator set. Although maybe it's worth adding it and extending the messages, as you suggest 🤔

Ahhh right, I see! So validator X and Y aren't necessarily either you :D.
I think this is fine, and sensible :). We then get a log of all the different validators which run an outdated version before they then drop out of consensus next epoch.

From a validator runner's perspective, there is a rather big difference between some other validator is out of date (I don't care), and your node is out of date (I do care). But as mentioned, it will crash at commit anyway :) - so I think that's the better place for this log message, as we're doing.

Reply:
Basically, I'm happy with your above message / proposals.